x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -598,54 +599,71 @@ public static class EnableSpatialDistancePushdown extends PhysicalOptimizerRules
         protected PhysicalPlan rule(FilterExec filterExec, LocalPhysicalOptimizerContext ctx) {
             PhysicalPlan plan = filterExec;
             if (filterExec.child() instanceof EsQueryExec) {
-                if (filterExec.condition() instanceof EsqlBinaryComparison comparison) {
-                    ComparisonType comparisonType = ComparisonType.from(comparison.getFunctionType());
-                    if (comparison.left() instanceof StDistance dist && comparison.right().foldable()) {
-                        plan = rewriteComparison(filterExec, dist, comparison.right(), comparisonType);
-                    } else if (comparison.right() instanceof StDistance dist && comparison.left().foldable()) {
-                        plan = rewriteComparison(filterExec, dist, comparison.right(), ComparisonType.invert(comparisonType));
+                List<Expression> rewritten = new ArrayList<>();
+                List<Expression> notRewritten = new ArrayList<>();
+                for (Expression exp : splitAnd(filterExec.condition())) {
```

Comment:
Why is this only checking a combination of `and` statements?
Does this mean the pushdown happens only when there is a `where  cond1 and cond2 and cond3....`? Does it make sense for other types of boolean conditions? `not`, `or`.

Reply:
Good point. I based this on the main loop of `PushFiltersToSource`, which only looks at conjunctive outer expressions. But on further investigation, I see that deeper down it does special cases for negation (line 283), and disjunction (line 279). I think I could expand to cover these cases in a followup PR.

Reply:
Do we normalize to conjunctive normal form in the logical planner, so that doing this top-level `splitAnd` in the physical planner always works?

Reply:
No. In the optimizer these conjunctions and disjunctions are optimized from a different point of view.

Reply:
Added an issue to capture this missing logic: https://github.com/elastic/elasticsearch/issues/110349